### PR TITLE
Feature/sync facility profile

### PR DESCRIPTION
**Pull request by jofomah**

### CODE REVIEW

File: app/scripts/services/sync-service.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  var getLocalDB = function(dbUrl){
+    return pouchdb.create(dbUrl);
+  };
+
+  var getRemoteDB = function(dbName){
+    var REMOTE = config.api.url + '/' + dbName;
+    return pouchdb.create(REMOTE);
+  };
+
+  var insertItem = function(db, item){
```
Comment (tlvince) [inline app/scripts/services/sync-service.js]: `db.put` already returns a promise/is wrapped by `$q`; see [tlvince/angular-pouchdb#qify](https://github.com/tlvince/angular-pouchdb/blob/db6fae6bc871c175c169d9d61a43c8800cde836c/angular-pouchdb.js#L5-L23).

Couldn't you just `return db.put(item, item.uuid)`?

File: app/scripts/services/sync-service.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      var remoteDB = getRemoteDB(dbName);
+      var localDB = getLocalDB(dbName);
+      isSyncing = true;
+
+      var onSyncComplete = function (deferred) {
+        isSyncing = false;
+        deferred.resolve(true);
+      };
+
+      updatePouchDBWithItem(localDB, item).then(function (result) {
```
Comment (tlvince) [inline app/scripts/services/sync-service.js:54]: Wrapping the replication methods in `$q` would make a good patch for [tlvince/angular-pouchdb](https://github.com/tlvince/angular-pouchdb/blob/db6fae6bc871c175c169d9d61a43c8800cde836c/angular-pouchdb.js#L54).

Comment (jofomah) [inline app/scripts/services/sync-service.js:54]: so should i add this to angular-pouchdb and send it as a patch??

File: app/scripts/services/sync-service.js (hunk 1 above)
Comment (jofomah) [inline app/scripts/services/sync-service.js]: my initial idea was to abstract common feature but you are right they are almost same, well still cleaning up syncService, btw, i think we should abstract syncing from stock count to a sync-service, what do you think??

**Pull request merged**